### ISSUE TITLE
Airplay Transmitter Feature Request

### ISSUE THREAD

**Issue by PhoGGy** [mikebrady/shairport-sync#120]
Hey Mike!
Thanks for your great work with shairport-sync, and all the others before with shairport!
I have a feature request or at least i would like to hear from you if it would be possible:
I want to stream music via Airplay to one raspberry pi. This Pi shall play the music, but redirect it to another Pi, too.
In other words, would it be possible to combine your pipe output with forked-daapd or pulseAudio or whatever to create a multiroom experience.
I´m just an advanced user of all this, so im not the guy who can code this.
You think its possible?

Thanks!

Greetings, Phil

**Comment by mikebrady** [comment]
Hi Phil. Thanks for your kind words.
There has been a number of suggestions similar to yours, where Shairport Sync would receive an audio stream from a source and then distribute it synchronously to other Shairport Sync endpoints, each playing to a loudspeaker.

It seems to me that there would be quite a lot of work involved in this extra feature -- you'd have to build a complete transmitting system, capable of outputting to a number of streams simultaneously. It would need to be able to do synchronisation, handle missing audio sinks, etc. gracefully and so on.

The main reason for not doing this is that its functionality is already in place in iTunes -- you can output to multiple speakers (including multiple Shairport Sync speakers) simply by adding them to the selection of speakers. The feature is not present in iOS, however, presumably due to processing overhead. Even so, there are third-party apps, such as Whaale, that do provide multi speaker streaming.

So, I hope you'll understand that I think it would not really be worth the considerable extra effort.

Kind regards
Mike

**Comment by PhoGGy** [comment]
Hey Mike!
Thank you for your response!
As far as I found out now it should be possible fairly easy. 
I'm trying to compile the pulseaudio-raop2 sources at the moment.
pulseaudio offers the possibility to take a pipe or the alsa output as its input.
Then, with the raop2 plugin properly compiled, it can output to another AirPlay station.
My approach so far would be to take your shairport-sync and let it output normally via alsa to the headphone jack. At the same time let pulseaudio grab the sound from the alsa output and transmit it to a fixed(hardcoded) AirPlay station. 
At the moment I'm stuck at compiling the raop2 properly, but I'll get it to work soon, I think. 
To sum it up, I think it's easily possible to stream to a hard coded station. Of course it's a very unstable solution so far but better than nothing! ;-)
I want to get it done with iOS without any extra app. 

Greetings Phil.

**Comment by mikebrady** [comment]
Thanks for the update -- please keep us posted about your progress!

**Comment by benhornshaw** [comment]
Did you get Pulseaudio-raop2 to work Phil?
I am struggling to make it work myself at the moment.
I have a slightly different purpose. but along the same lines. I use Shairport to receive Audio. But i use MPD/MPC to play back radio streams and its those that i would like to send over pulseaudio.
Any Guidance would be greatly received.

Thanks Ben
